crates/sui-gateway/src/event_api.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::fmt::Display;
+use std::sync::Arc;
+
+use futures::{StreamExt, TryStream};
+use jsonrpsee_core::error::SubscriptionClosed;
+use jsonrpsee_core::server::rpc_module::{PendingSubscription, SubscriptionSink};
+use jsonrpsee_proc_macros::rpc;
+use move_core_types::parser::parse_struct_tag;
+use serde::Serialize;
+
+use sui_core::authority::AuthorityState;
+use sui_core::event_filter::EventFilter;
+use sui_core::event_handler::EventHandler;
+use sui_core::gateway_types::SuiEvent;
+
+#[rpc(server, client, namespace = "sui")]
+pub trait EventApi {
+    #[subscription(name = "subscribeMoveEventsByType", item = SuiEvent)]
+    fn subscribe_move_event_by_type(&self, event: String, field_filter: BTreeMap<String, String>);
+}
+
+pub struct EventApiImpl {
+    state: Arc<AuthorityState>,
+    event_handler: Arc<EventHandler>,
+}
+
+impl EventApiImpl {
+    pub fn new(state: Arc<AuthorityState>, event_handler: Arc<EventHandler>) -> Self {
+        Self {
+            state,
+            event_handler,
+        }
+    }
+}
+
+impl EventApiServer for EventApiImpl {
+    fn subscribe_move_event_by_type(
+        &self,
+        pending: PendingSubscription,
+        event: String,
+        field_filter: BTreeMap<String, String>,
+    ) {
+        let event_type = match parse_struct_tag(&event) {
```

Comment:
Can you document what the input to `parse_struct_tag` looks like, what it means?